alpha/lib/model/DeliveryProfilePeer.php

```diff
@@ -872,5 +872,13 @@ public static function getCustomDeliveryProfileIds($entry, $partner, DeliveryPro
 
 		return $deliveryIds;
 	}
+
+	public static function retrieveByPartnerAndSystemName($partnerId, $systemName)
```

Comment:
Since there is no use for multiple delivery profile returned here, better use doSelectOne, and that can reduce code in callers to avoid taking the first item (reset(retrieveByPartnerAndSystemName(...)).